### DISCUSSION TITLE
Suggestions for getDefinitionsProductType API

### DISCUSSION THREAD

**Discussion by wszwc** [amzn/selling-partner-api-models#577] [Ideas]
It is strongly recommended to rectify the json schema returned by the getDefinitionsProductType API. It is too difficult to convert the current product type JsonSchema into an HTML UI form that can be filled in by users. I think this is completely unnecessary. The following is the problem:
1. Amazon's backend has three form modes, required mode, recommended mode, and all attributes. The JsonSchema of the product type does not have this part of the content, which makes it very difficult for users to successfully submit a product through my HTML UI form. Difficult simply because too many unnecessary fields interfere with
2. The Variation_theme field returns all themes, including variation themes that do not belong to the current product type. This is a very bad thing for me to deal with multi-variation product listings and has no benefit at all. And I have no basis for requiring users to fill in the attributes associated with Variation_theme. Many attributes with enumeration values have this problem.
3. I don’t understand why there are such attributes as other_offer_image_locator_1, 2, 3, and 4 since you have defined all attributes as arrays.

**Comment by vaidas-lungis** [comment]
Fully agree  +1

**Reply by wszwc** [reply]
I can't believe this is an obvious and urgent question, why can't any Amazon developer answer it?
